Sources/Sentry/SentryNSDataTracker.m

```diff
@@ -173,9 +207,43 @@ - (NSData *)measureNSDataFromURL:(NSURL *)url
 
     [ioSpan setDataValue:path forKey:@"file.path"];
 
+    [self mainThreadExtraInfo:ioSpan];
+
     return ioSpan;
 }
 
+- (void)mainThreadExtraInfo:(id<SentrySpan>)span
+{
+    BOOL isMainThread = [NSThread isMainThread];
+
+    [span setDataValue:@(isMainThread) forKey:@"blocked_main_thread"];
```

Comment:
`h`: This change significantly impacts the payload size. Can't we only send this if the span blocked the main thread? Why do we need to add this data value if it didn't block the main thread?

Reply:
Is just a boolean. How this `significantly impacts the payload size`?
I guess we could not include it, but this is how was decided in the RFC and how android do it.

@romtsn thoughts?

Reply:
imo it's fine to always send the boolean, I don't think the payload size will suffer much from this (we also agreed with backend to always send it). More important is that we should not send the stacktrace if the boolean is `false`, but this is already considered on the line 221.

Reply:
> I don't think the payload size will suffer much from this

For every file I/O span, we now will send `blocked_main_thread`: `false`. Such small things all add up to a payload size increase. IMO the default should be `false`, and we should only send it if `blocked_main_thread`: `true`.